src/clj_kondo/impl/namespace.clj

```diff
@@ -207,8 +207,11 @@
 
 (defn reg-alias!
   [{:keys [:base-lang :lang :namespaces]} ns-sym alias-sym aliased-ns-sym]
-  (swap! namespaces assoc-in [base-lang lang ns-sym :qualify-ns alias-sym]
-         aliased-ns-sym))
+  (swap! namespaces
+         (fn [n]
+           (-> n
+               (assoc-in [base-lang lang ns-sym :qualify-ns alias-sym] aliased-ns-sym)
+               (assoc-in [base-lang lang ns-sym :aliases alias-sym] aliased-ns-sym)))))
```

Comment:
This can be made more efficient by using `(update-in [base-lang lang ns-sym] assoc :qualify-ns ... :aliases ...)`

Reply:
@borkdude sorry, I'm having trouble seeing this, maybe I'm missing something obvious. It's tricky because `alias-sym` is a key on siblins. Are you suggesting something like this?

```
(update-in [base-lang lang ns-sym] 
  #(-> % 
    (update :qualify-ns assoc alias-sym aliased-ns-sym)
    (update :aliases assoc alias-sym aliased-ns-sym)))
```

Reply:
You're right, that gets convouted, let's leave it.